server/sonar-server/src/main/java/org/sonar/server/source/ws/ShowAction.java

```diff
@@ -82,33 +83,37 @@ public void define(WebService.NewController controller) {
   @Override
   public void handle(Request request, Response response) {
     String fileKey = request.mandatoryParam("key");
-    userSession.checkComponentPermission(UserRole.CODEVIEWER, fileKey);
-
-    int from = Math.max(request.mandatoryParamAsInt("from"), 1);
+    int from = Math.max(request.paramAsInt("from"), 1);
     int to = (Integer) ObjectUtils.defaultIfNull(request.paramAsInt("to"), Integer.MAX_VALUE);
 
-    DbSession session = dbClient.openSession(false);
+    DbSession dbSession = dbClient.openSession(false);
     try {
-      ComponentDto componentDto = componentFinder.getByKey(session, fileKey);
-      List<String> linesHtml = sourceService.getLinesAsHtml(componentDto.uuid(), from, to);
-      JsonWriter json = response.newJsonWriter().beginObject();
-      writeSource(linesHtml, from, json);
+      ComponentDto file = componentFinder.getByKey(dbSession, fileKey);
+      userSession.checkProjectUuidPermission(UserRole.CODEVIEWER, file.projectUuid());
+
+      Optional<Iterable<String>> linesHtml = sourceService.getLinesAsHtml(dbSession, file.uuid(), from, to);
+      if (linesHtml.isPresent()) {
+        JsonWriter json = response.newJsonWriter().beginObject();
+        writeSource(linesHtml.get(), from, json);
+        json.endObject().close();
+      } else {
+        throw new NotFoundException();
+      }
 
-      json.endObject().close();
     } finally {
-      session.close();
+      dbClient.closeSession(dbSession);
     }
-
   }
 
-  private void writeSource(List<String> lines, int from, JsonWriter json) {
+  private void writeSource(Iterable<String> lines, int from, JsonWriter json) {
     json.name("sources").beginArray();
-    for (int i = 0; i < lines.size(); i++) {
-      String line = lines.get(i);
+    int index = 0;
+    for (String line : lines) {
       json.beginArray();
-      json.value(1L * i + from);
+      json.value(index + from);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Cast one of the operands of this addition operation to a "long". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2184)